src/commands/utils/uploadUDF.ts

```diff
@@ -0,0 +1,123 @@
+} from "../../clients/flinkArtifacts";
+import { logError } from "../../errors";
+import { EnvironmentId, IEnvProviderRegion } from "../../models/resource";
+import {
+  showErrorNotificationWithButtons,
+  showInfoNotificationWithButtons,
+  showWarningNotificationWithButtons,
+} from "../../notifications";
+import { getSidecar } from "../../sidecar";
+
+/**
+ * Prompts the user for environment, cloud provider, region, and artifact name.
+ * Returns an object with these values, or undefined if the user cancels.
+ */
+export interface UDFUploadParams {
+  environment: string;
+  cloud: string;
+  region: string;
+  artifactName: string;
+  fileFormat: string;
+}
+
+/**
+ * Requests a presigned upload URL for a Flink artifact using the sidecar.
+ * @param request PresignedUploadUrlArtifactV1PresignedUrlRequest
+ * @returns The presigned URL response object, or undefined if the request fails.
+ */
+export async function getPresignedUploadUrl(
+  request: PresignedUploadUrlArtifactV1PresignedUrlRequest,
+): Promise<PresignedUploadUrlArtifactV1PresignedUrl200Response | undefined> {
+  try {
+    const sidecarHandle = await getSidecar();
+    const providerRegion: IEnvProviderRegion = {
+      environmentId: request.environment as EnvironmentId,
+      provider: request.cloud,
+      region: request.region,
+    };
+    const presignedClient = sidecarHandle.getFlinkPresignedUrlsApi(providerRegion);
+
+    // Wrap the request as required by the OpenAPI client
+    const urlResponse = await presignedClient.presignedUploadUrlArtifactV1PresignedUrl({
+      PresignedUploadUrlArtifactV1PresignedUrlRequest: request,
+    });
+    return urlResponse;
+  } catch (err) {
+    logError(err, "Failed to get presigned upload URL");
+    return undefined;
+  }
+}
+
+/**
+ * Handles the presigned URL request and shows appropriate notifications.
+ * @param request The presigned upload URL request
+ */
+export async function handlePresignedUrlRequest(
+  request: PresignedUploadUrlArtifactV1PresignedUrlRequest,
+): Promise<void> {
+  const response = await getPresignedUploadUrl(request);
+  if (response && response.upload_url) {
+    showInfoNotificationWithButtons("Presigned upload URL received.", {
+      "Copy URL": () => {
+        vscode.env.clipboard.writeText(response.upload_url!);
+        vscode.window.showInformationMessage("Upload URL copied to clipboard.");
+      },
+    });
+  } else {
+    showErrorNotificationWithButtons("Failed to get presigned upload URL. See logs for details.");
+  }
+}
+export async function promptForUDFUploadParams(): Promise<UDFUploadParams | undefined> {
+  const environment = await vscode.window.showInputBox({
+    prompt: "Enter the Environment ID for the UDF upload",
+    ignoreFocusOut: true,
+    validateInput: (value) => (value ? undefined : "Environment ID is required"),
+  });
+  if (!environment) {
+    showErrorNotificationWithButtons("Upload UDF cancelled: Environment ID is required.");
+    return undefined;
+  }
```

Comment:
We should replace this with [`flinkCcloudEnvironmentQuickPick()`](https://github.com/confluentinc/vscode/blob/88b716e5803598b932fcc70fd33ee4d8d33b31c0/src/quickpicks/environments.ts#L60) for a better user experience. Then we should get a `CCloudEnvironment` or `undefined` and either pull the `id` or exit early